cognite/client/_api/data_modeling/instances.py

```diff
@@ -244,10 +244,10 @@ def __call__(
         )
 
         if instance_type == "node":
-            resource_cls: type = _NodeOrEdgeResourceAdapter
+            resource_cls: type = _NodeOrEdgeResourceAdapter(Node, Edge)  # type: ignore[assignment]
             list_cls: type = NodeList
         elif instance_type == "edge":
-            resource_cls, list_cls = _NodeOrEdgeResourceAdapter, EdgeList
+            resource_cls, list_cls = _NodeOrEdgeResourceAdapter(Node, Edge), EdgeList  # type: ignore[assignment]
```

Comment:
Perhaps a silly question but why do we need to use an adapter when we know the instance type?

```

if instance_type == "node":
    resource_cls, list_cls = Node, NodeList
elif instance_type == "edge":
    resource_cls, list_cls = Edge, EdgeList
```